### DISCUSSION TITLE
Requests：閲覧権限設定に、自分を含んでいないグループを含めて複数設定したい。

### DISCUSSION THREAD

**Discussion by funakura** [growilabs/growi#9091] [Ideas / Requests] [answered]
現在は、自分を含んだグループのみ選択可能です。
これでは、他部門(自分は含まない)と情報共有する場合に対応できません。

以下の３グループがあった場合
部門１（メンバ：A,B,C）
部門２（メンバ：X,Y,Z）
部門３（メンバ：α,β,γ）

複数のグループ設定が出来ても、Aさんは次の権限設定ができません。
「部門１」amd「部門２」

自分を含まないグループも指定したいです。
グループの存在自体を見せたくない場合の対応などありますが、検討いただけると助かります。

または、以下のような重複した親子関係が作れると管理が楽になります。（組み合わせが爆発しますが...）
親「部門１＆部門２」->子「部門１」「部門２」
親「部門１＆部門３」->子「部門１」「部門３」

以上、検討よろしくお願いします。

**Comment by yuki-takei** [comment, marked as answer]
現状、以下で部門1と部門2両方のグループで共有できるページ群を作成することは可能です。

# ワークアラウンド
1. 部門1と部門2両方に所属するユーザーを一時的に作成する(ユーザーXとする)
2. ユーザーXで `/部門1,2共通` というパス(適当)のページを作り、両グループによる読み書きができるよう設定する
3. (ユーザーXを削除する)

`/部門1,2共通` の子ページを作成する時に、親ページのグループ設定をそのまま踏襲するか、または作成者が改めてグループを設定するのかを選択できます。改めてグループを設定する場合は、部門1と部門2またはその子グループ かつ 作成者が所属するグループのみを選択できます。

組み合わせの爆発に対しては、一旦ユーザーXを全グループに所属させてから
- `/部門1,2`
- `/部門2,3`
- `/部門1,3`
- `/部門1,2,3`
- ...
のようにページツリー内での最上位ページを予め量産することでグループの順列組み合わせ的管理は免れることができます。

# Note
`/部門1,2共通` のページ配下に部門3グループがアクセスできるページを作成する機能は現状搭載する予定はありません。
(ページ群の移動や削除等に対応できなくなるため)

**Comment by funakura** [comment]
解決策の提案ありがとうございます。
仕様理解できました。

**Comment by funakura** [comment]
時間が空きましたが試してみました。
親ページのグループ設定をそのまま踏襲した場合に、あとで自分が所属したいグループのみに絞ることができません。

現在：部門1(作成者所属)と、部門2(作成者未所属)にアクセス権が付いている。

グループの再設定：部門1(作成者所属)のみにしたい。
部門1(作成者所属)はON/OFFできる
しかし、部門2(作成者未所属)が選択されたままでグレーアウト

なかなな難しいですね。

| item     | version |
| ---      | --- |
|OS        ||
|GROWI     |7.0.16|
|node.js   |20.16.0|
|npm       |10.8.1|
|yarn      |1.22.22|
